lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -1470,6 +1523,9 @@ class DedupPass : public DedupBase<DedupPass> {
     if (failed)
       return signalPassFailure();
 
+    for (auto module : modules)
+      AnnotationSet::removeAnnotations(module, dedupGroupClass);
```

Comment:
It seems `modules` is already invalid here. Probably it should be sufficient to just replace `module` with `circuit.getOps<FModuleOp>()`.